app/src/main/java/org/schabi/newpipe/local/dialog/PlaylistAppendDialog.java

```diff
@@ -26,8 +26,12 @@
 public final class PlaylistAppendDialog extends PlaylistDialog {
     private static final String TAG = PlaylistAppendDialog.class.getCanonicalName();
 
+    private static final float DEFAULT_ALPHA = 1f;
+    private static final float GRAYED_OUT_ALPHA = 0.3f;
+
     private RecyclerView playlistRecyclerView;
     private LocalItemListAdapter playlistAdapter;
+    private List<Long> duplicateIds;
```

Comment:
Mmmh, I don't like this approach. It is slow, since at the moment queries run in `O(n)`, while with this code showing the list would require `O(n²)` time (where `n` is the number of playlists), since for each playlist being shown you need to check if its id is among those duplicate. Can't you just return that information in the `getPlaylists` database query, after changing it to take the video id as parameter (or -1 in order not to show duplicate information in other places where playlists are shown)? You could add a `boolean wouldVideoBeDuplicate` field to `PlaylistMetadataEntry` items, that is not stored in the database but is instead returned only in queries. Then you can use that field in local item views to change the alpha of the whole item view.

Reply:
Thanks for your input. However, after having another look, I have another suggestion.
When you refer to the `O(n^2)` time required, you probably talk about the the checks in the `showDuplicateIndicators` method. If I adjust this method a bit, the loop within this method is only repeated a maximum of 10 times (only checking the currently visible playlists). In this loop, their Id gets only compared to the duplicates, which is usually a really low number (`duplicates << playlists`), since 99% of people won't add one video to all of their playlists.

Furthermore, the current implementation would be faster if the are no duplicates at all (which is the large majority of all requests), right? Since you can simply check if the list is empty before you go through any of the playlist items.
IMO, with a simple change this would be performant and keep us from changing the `getPlaylist` query and adding another field to the `PlaylistMetadataEntry`.

What's your opinion? Is there anything I missed?

Reply:
I pushed a commit containing what I mean. It would have been harder to explain than to do ;-)

I agree that your approach wouldn't really be slow, but it also has to access the list from various unusual places. The solution I pushed I think is more neat.

Reply:
I just looked at you changes and agree that your solution is preferable. However, at the moment there is no text that explains what a grayed out playlist actually means. I initially created an invisible explainer text (`NewPipeTextView`) that you can set to visible if there are duplicates. Did you think this was unnecessary or did you just forget to include it in your current solution approach?

Reply:
I forgot to include it, sorry

Reply:
Ok, I can add this and your other suggestions. Just one more question: Do you think using a for loop until you find the first one with a duplicate is performant enough? Since `PlaylistDuplicatesEntry` has the field `timesStreamIsContained`, it should be in O(N) (worst case). And since this field is not stored in the database, a query wouldn`t be faster as well.

Reply:
Yeah, that's fine, I doubt there is any clever trick (and also O(n) is perfectly ok and does not increase complexity if the data has already been collected in O(n) anyway)